### ISSUE TITLE
[Bug]: Training Error -- Need At Least 1 Array to Concatenate Issue

### ISSUE THREAD

**Issue by NitrogenSulfide** [IAHispano/Applio#1119]
### Project Version

3.3.0

### Platform and OS Version

W11

### Affected Devices

RTX 5090 32GB

### Existing Issues

https://github.com/IAHispano/Applio/issues/1020

### What happened?

I am able to get to Step 4 AKA the Training but then when I click "Start Training" it will run like 5 seconds before stopping and this is what I have from the logs:

`Starting pitch extraction on cuda:0 using rmvpe...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 394/394 [00:17<00:00, 22.01it/s]
Pitch extraction completed in 21.53 seconds.
Starting embedding extraction with 16 cores on cuda:0...
Embedding extraction completed in 2.95 seconds.
Not enough data present in the training set. Perhaps you forgot to slice the audio files in preprocess?
An error occurred extracting the index: need at least one array to concatenate
If you are running this code in a virtual environment, make sure you have enough GPU available to generate the Index file.`

Other info:

I did look at https://github.com/IAHispano/Applio/issues/1020 which was in April of 2025, and I executed the pytorch commands using the person's /env/python. That still didn't help. 

### Steps to reproduce

1. Go to Training tab, named the new model as usual, sampling rate at 48000 and CPU cores at 16 which is default. GPU is index 0 -- RTX 5090. This step seemingly runs without issue.
2. Preprocess section -- Point dataset (auto selects already from dropdown) to correct location with about 400 1 - 8 second high quality 768kbps and 48khz wav files. These are all cleaned and formatted already. SKIP for audio cutting, I add noise filter, and "none" for normalization mode, and no noise reduction. Output says the model preprocessed successfully. Good so far.
3. Extract -- default rmvpe and contentvec. I put 0 for silent training files but also tried with default "2" value. Successful still.
4. Training -- Batch size, I tried 4 all the way up to 24, save every epoch and total epoch default values. I agree to terms of use and click "Start Training" -- I run into the above logs. Generate index before or after doesn't do anything different either.

### Expected behavior

It should have started training without any errors.

### Screenshots or Videos

<img width="3809" height="293" alt="Image" src="https://github.com/user-attachments/assets/e40d3cc7-3ca5-4d8a-829f-1b1e05f451b3" />

**Comment by AznamirWoW** [comment]
btw, you did not need to run anything for 5090 support, that fix is included in v3.3.0

**Comment by AznamirWoW** [comment]
rename rvc\train\extract\extract.py to extract.old, then drop the attached file into the same folder. Restart Applio and try running extract features again. It should show an error message if there's any.

[extract.py](https://github.com/user-attachments/files/21847586/extract.py)

**Comment by NitrogenSulfide** [comment]
> rename rvc\train\extract\extract.py to extract.old, then drop the attached file into the same folder. Restart Applio and try running extract features again. It should show an error message if there's any.
> 
> [extract.py](https://github.com/user-attachments/files/21847586/extract.py)

Thanks for the new file. I didn't see anything new in the logs:

`An error occurred connecting to Discord: Could not find Discord installed and running on this machine.
Failed to launch on port 6969, trying again on port 6968...
Failed to launch on port 6968, trying again on port 6967...
Failed to launch on port 6967, trying again on port 6966...
* Running on local URL:  http://127.0.0.1:6966

To create a public link, set `share=True` in `launch()`.
Starting preprocess with 16 processes...
100%|██████████████████████████████████████████████████████████████████████████████████| 394/394 [00:08<00:00, 48.96it/s]
Preprocess completed in 8.05 seconds on 00:40:28 seconds of audio.
Starting pitch extraction on cuda:0 using rmvpe...
100%|█████████████████████████████████████████████████████████████████████████████| 1097/1097 [00:00<00:00, 21320.38it/s]
Pitch extraction completed in 3.70 seconds.
Starting embedding extraction with 16 cores on cuda:0...
Embedding extraction completed in 2.97 seconds.
Not enough data present in the training set. Perhaps you forgot to slice the audio files in preprocess?
An error occurred extracting the index: need at least one array to concatenate
If you are running this code in a virtual environment, make sure you have enough GPU available to generate the Index file.`

Maybe I should try a manual git install of Applio? into a separate folder cleanly, and try again?

**Comment by AznamirWoW** [comment]
> Maybe I should try a manual git install of Applio? into a separate folder cleanly, and try again?

Don't think that's the problem.

Normal extract feature should look like this:

<img width="1905" height="126" alt="Image" src="https://github.com/user-attachments/assets/ab3f0700-1308-48d5-9847-757ce513867a" />

Can you share your dataset?

**Comment by AznamirWoW** [comment]
> I have shared an example file I am using. There are about 300-400 other files with the exact same properties, except for length. The length will vary between 1 to 8 seconds.

If you want me to try and reproduce your issue, I need your whole set.

Generally I do not recommend using loose files as they come with different duration. Ideally you want a uniformly sliced dataset with some overlaps between segments.

**Comment by AznamirWoW** [comment]
> Ah ok, is that what the preprocess slicing is for? I noticed the default is 3 seconds, and I see that longer wav files are cut up so sentences are cut off etc. Is that ok?

preprocess slicing accepts large files (10+ minutes) and uses either a new simple method or old automatic slicing method.
Simple method is very fast and it assumes you've prepared your dataset file and removed all excessive silences >0.25s.
Automatic attempts to find gaps between words and use that as a slice points.

Simple slicing produces uniform 3s slices with 0.3s overlap.
Automatic method may produce the same 3s slices and overlap if it finds no silences, or 1-5s segments if it does.

Using segments smaller and 3s or larger than 5s is not recommended.

**Comment by idolistmedalist** [comment]
I did exactly the same as described in the problem on rented servers 3090/4090 on different versions of cuda, before everything worked fine, but now after installing this:

![Image](https://github.com/user-attachments/assets/b1e68c75-58d3-4cce-bfc8-5d4a7c08d2e4)

**Comment by idolistmedalist** [comment]
> > I did exactly the same as described in the problem on rented servers 3090/4090 on different versions of cuda, before everything worked fine, but now after installing this:
> > ![Image](https://github.com/user-attachments/assets/b1e68c75-58d3-4cce-bfc8-5d4a7c08d2e4)
> 
> Did you install it manually? Or did you use a compiled version?

Manually, everything worked fine before, now I tried it on cuda 12.6 and this comes up.

![Image](https://github.com/user-attachments/assets/4df187e4-4dd3-461f-a893-cc0bbaa8a3a0)

**Comment by idolistmedalist** [comment]
> <img alt="Image" width="294" height="169" src="https://private-user-images.githubusercontent.com/133521603/479545648-42f909bd-66ce-42fc-996c-e5c05a5cd61e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU2MTM2NzUsIm5iZiI6MTc1NTYxMzM3NSwicGF0aCI6Ii8xMzM1MjE2MDMvNDc5NTQ1NjQ4LTQyZjkwOWJkLTY2Y2UtNDJmYy05OTZjLWU1YzA1YTVjZDYxZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxOVQxNDIyNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjYxZjkwOGJjMmYzYWNlOGVlZmRjNDEzOTAwZDNiYjYzZmQ3NTgwNDIwNjFlY2RkNzQ5NzI2N2E1ZDA0NTQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lw1yX58bEOt57n61hwOP2mUAcN7FlsaOxRSxA9uNuig">
> i see two OS here
I rent servers on 3090/4090 on linux my pc has a very long training time just
